scapy/tools/automotive/isotpscanner.py

```diff
@@ -0,0 +1,157 @@
+               "python2 -m scapy.tools.automotive.isotpscanner "
+               "\"can.interface.Bus(bustype='vector', channel=0, "
+               "bitrate=250000)\" 0 100 \n"
+               "python2 -m scapy.tools.automotive.isotpscanner "
+               "\"can.interface.Bus(bustype='socketcan', channel='can0', "
+               "bitrate=250000)\" 0 100 \n\n"
+               "Python3 on Linux:\n"
+               "python3 -m scapy.tools.automotive.isotpscanner can0 0 100")
+    parser.add_argument("interface", type=str,
+                        help="CAN interface for the scan.\n"
+                             "Depends on used interpreter and system,\n"
+                             "see examples below. Any python-can interface can"
+                             " be provided. Please see: "
+                             "https://python-can.readthedocs.io for further "
+                             "interface examples.")
+    parser.add_argument("startID", type=lambda x: int(x, 16),
+                        help="Start scan at this identifier (hex)"),
+    parser.add_argument("endID", type=lambda x: int(x, 16),
+                        help="End scan at this identifier (hex)")
+    parser.add_argument("-n", "--noise_listen_time", type=int, default=2,
+                        help="Set seconds listening for noise before scan.")
+    parser.add_argument("-t", "--sniff_time", type=int, default=100,
+                        help="Set the duration, a sniff is waiting for a "
+                             "flow-control response. In milliseconds.")
+    parser.add_argument("-e", "--extended", action="store_true",
+                        help="Include ISOTP extended addressing to scan.")
+    parser.add_argument("-eo", "--extended_only", action="store_true",
+                        help="Scan only with ISOTP extended addressing.")
+    parser.add_argument("-p", "--piso", action="store_true",
+                        help="Print 'Copy&Paste'-ready ISOTPSockets.")
+    parser.add_argument("-v", "--verbose", action="store_true",
+                        help="Display information during scan.")
+
+    args = parser.parse_args()
+
+    scan_interface = args.interface
+    interface_string = ""
+    if "can.interface.Bus" in scan_interface:
+        if PYTHON_CAN:
+            import can  # noqa: 401
+            try:
+                scan_interface = eval(scan_interface)
+                interface_string = "CANSocket(%s)" % args.interface
+            except Exception as e:
+                print("Check your interface string.\n"
+                      "ISOTPScanner.py -h for usage examples.\n")
+                print(e)
+                exit(-1)
+        else:
+            print("Wrong interface type.\n"
+                  "ISOTPScanner.py -h for usage examples.")
+            exit(-1)
+    else:
+        if PYTHON_CAN:
+            print("Wrong interface type.\n"
+                  "ISOTPScanner.py -h for usage examples.")
+            exit(-1)
```

Comment:
`exit` is usually reserved for command line. You would use `sys.exit` within code

Reply:
thanks